woocommerce/woocommerce-gutenberg-products-block
# Migrate REST API usage to the Store API

Upon reviewing the REST API overrides in blocks, to see what needed to be ported to the REST API itself, I found that most endpoints simply needed to expand access capabilities to authors/non-admin users. 

As well as expanding access rights, some of the endpoints were then needing to shape data to hide private things, or return data in a specific opinionated format. This didn't seem like a good solution to me, so I worked through each endpoint to see if the **Store API** could be used instead. Store API is public, so there should be no auth issues in doing this.

I had success with the existing Store endpoints (products, attributes) so decided to port over some of the other endpoints too, creating new Store API endpoints for tags, categories, reviews all following similar practices and only allowing READ access to public data.

This PR contains that work :)

- Introduces endpoints for tags, categories, reviews
- Expands some endpoints with missing data and params
- Removes all REST API overrides
- Refactors some HOCS which used Rest API
- Switches all APIFetches to Store API

### How to test the changes in this Pull Request:

1. Test a variety of blocks including reviews by x, featured product, products by attribute/tag/category.
2. Confirm all products block works
3. Double check cart/checkout is still functional. I didn't touch too much there but did move some code around.
